upstream/clients.go

```diff
@@ -92,6 +93,12 @@ func (manager *ClientRegistry) CreateClient(ups *Upstream) (ClientInterface, err
 					clientErr = fmt.Errorf("failed to create Pimlico client for upstream: %v", cfg.Id)
 				}
 
+			case common.UpstreamTypeEvmEtherspot:
+				newClient, err = NewEtherspotHttpJsonRpcClient(ups, parsedUrl)
```

Comment:
We also need a condition in upstream.go "Upstream.guessUpstreamType()" for cases when type is not defined by the user.